### PR TITLE
feat: codeartifact deploy

### PR DESCRIPTION
**Pull request by loydbanks**

### CODE REVIEW

File: .github/workflows/codeartifact-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          role-duration-seconds: 600
+
+      - name: Set Up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
+
+      - name: Build Package
+        run: |
+          python setup.py sdist
```
Comment (paddatrapper) [inline .github/workflows/codeartifact-deploy.yml:35]: Don't call setup.py directly - you're removing it as part of the move to pyproject.toml

File: .github/workflows/codeartifact-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Publish Package
```
Comment (paddatrapper) [inline .github/workflows/codeartifact-deploy.yml:36]: This task doesn't make sense - you're installing packages from the repo, not publishing to it

File: .github/workflows/codeartifact-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        run: |
+          aws codeartifact login --domain BEAM --repository BEAM Private Packages django-helpdesk &&
+          pip install -r requirements.txt
+      - name: Run Tests
+        run: |
+          pip install -r requirements-testing.txt
+          python3 quicktest.py
```
Comment (paddatrapper) [inline .github/workflows/codeartifact-deploy.yml:43]: ?

**Review by paddatrapper — CHANGES_REQUESTED** [review]

File: .github/workflows/codeartifact-deploy.yml (hunk 3 above)
Comment (loydbanks) [reply .github/workflows/codeartifact-deploy.yml:43]: According to the README file, this is how we run tests for this project.

Comment (paddatrapper) [reply .github/workflows/codeartifact-deploy.yml:43]: Why are you running tests at this point? This relates to https://github.com/ClearlyEnergy/django-helpdesk/pull/23#discussion_r1767059794

File: .github/workflows/codeartifact-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Install dependencies
+        run: |
+            python -m pip install --upgrade pip
+            pip install twine
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: us-east-2
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
```
Comment (paddatrapper) [inline .github/workflows/codeartifact-deploy.yml:54]: If you're using OIDC, then these are not required

**Review by paddatrapper — APPROVED** [review]

**Review by paddatrapper — APPROVED** [review]

**Pull request merged**